### ISSUE TITLE
Re-rendering the entire list of elements (ComboBox list)

### ISSUE THREAD

**Issue by EvgenPan** [adobe/react-spectrum#6981]
### Provide a general summary of the issue here

Every time you hover over  item in the list, all items are redrawn.

### 🤔 Expected Behavior?

Updating the item you hovered over.

### 😯 Current Behavior

Every time you hover over  item in the list, all items are redrawn. 
<img width="676" alt="Знімок екрана 2024-08-30 о 16 49 40" src="https://github.com/user-attachments/assets/9220b0c5-a66d-4f81-8e93-7b66b2a18fb6">

### 🖥️ Steps to Reproduce

-

### Version

version	1.3.3

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari

### What operating system are you using?

macOs

**Comment by LFDanLu** [comment]
This looks to be in the same vein as https://github.com/adobe/react-spectrum/issues/5897. Are you seeing a heavy performance hit?

**Comment by EvgenPan** [comment]
Yes, this becomes a huge problem when we have 2000 items @LFDanLu

**Comment by snowystinger** [comment]
Ah, you aren't virtualizing by any chance are you?

**Comment by EvgenPan** [comment]
@snowystinger Yes, sure.

**Comment by LFDanLu** [comment]
As @snowystinger alluded to, the performance hit can mitigated by virtualizing the combobox list, something we are hoping to support in a stable manner soon. For now, you can try using the UNSTABLE_Virtualizer like in this rough sandbox: https://codesandbox.io/p/sandbox/pedantic-black-vrd8rh

**Comment by EvgenPan** [comment]
@LFDanLu this solution doesn't work (

**Comment by LFDanLu** [comment]
@EvgenPan Mind expanding on what didn't work?
